### PR TITLE
TO: Hdr rw config test

### PR DESCRIPTION
**Pull request by moltzaum**
## What does this PR (Pull Request) do?

This PR adds a simple test for #3422. The number of lines are counted to make sure `__RETURN__` is replaced correctly.

While testing the generated config I found a divide by zero error and fixed it as well.

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?
Run the API tests

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR does not need documentation
- [x] This PR does not need an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by moltzaum** [comment]
@mitchell852

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3648/
Test PASSed.

File: traffic_ops/testing/api/v14/hdr_rw_dot_config_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+		return &ds
+	}
+
+	t.Errorf("Cannot test hdr_rw_dot_config with no http or dns deliveryservices: %s", err)
+	return nil
+
+}
+
+func getExpectedLines(rwRules string) int {
```
Comment (mitchell852) [inline traffic_ops/testing/api/v14/hdr_rw_dot_config_test.go]: this method will work for this:

getExpectedLines("rw1__RETURN__rw2__RETURN__edge__RETURN__header__RETURN__re-rewrite [L]")
=> returns 6

but won't work for this:

getExpectedLines("rw1" + "\n" + "re-rewrite [L]")
=> returns 2 when it should be 3

^^ think that's something you can fix so this method can be used for `__RETURN` and `\n`

File: traffic_ops/traffic_ops_golang/ats/headerrewrite.go [hunk 2]
```diff
@@ -73,12 +74,14 @@ func GetEdgeHeaderRewriteDotConfig(w http.ResponseWriter, r *http.Request) {
 			api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting ds server count: "+err.Error()))
 			return
 		}
-		maxOriginConnectionsPerEdge := int(math.Round(float64(maxOriginConnections) / float64(dsOnlineEdgeCount)))
-		text += "cond %{REMAP_PSEUDO_HOOK}\nset-config proxy.config.http.origin_max_connections " + strconv.Itoa(maxOriginConnectionsPerEdge)
-		if ds.EdgeHeaderRewrite == nil {
-			text += " [L]"
-		} else {
-			text += "\n"
+		if dsOnlineEdgeCount > 0 {
```
Comment (mitchell852) [inline traffic_ops/traffic_ops_golang/ats/headerrewrite.go]: nice catch! divide by zero...ug...

File: traffic_ops/testing/api/v14/hdr_rw_dot_config_test.go (hunk 1 above)
Comment (moltzaum) [reply traffic_ops/testing/api/v14/hdr_rw_dot_config_test.go]: I added a test case for this.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3662/
Test PASSed.

**Pull request merged**